Document GHC-01239 (closes #288) (#391)
haskellfoundation/error-message-index: message-index/messages/GHC-01239/example1/index.md

---
title: Unexpected if expression in function application
---

> To pass If-Then-Else expressions as function argument we either have to surround it in parentheses,

Reviewer: ```suggestion
To pass If-Then-Else expressions as function arguments, they must either be surrounded in parentheses,
```

Reviewer: Or if my suggested rewrite doesn't fit your taste, at least:
```suggestion
To pass If-Then-Else expressions as function arguments, we either have to surround them in parentheses,
```

Author: Went with the first one as I find it more natural.

> use function application operator `($)` or enable [`BlockArguments`](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/block_arguments.html#extension-BlockArguments) extension.

Reviewer: ```suggestion
used after the function application operator `($)`, or be used where the [`BlockArguments`](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/block_arguments.html#extension-BlockArguments) extension is enabled.
```

> use function application operator `($)` or enable [`BlockArguments`](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/block_arguments.html#extension-BlockArguments) extension.

Reviewer: I think the grammar is still a bit off. Here's a minimal fix:
```suggestion
use the function application operator `($)` or enable the [`BlockArguments`](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/block_arguments.html#extension-BlockArguments) extension.
```

Author: Thanks for pointing that out. I'm not native speaker and I always had a bit of an issue with articles :blush:
